src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -121,6 +121,15 @@ private String getCookie(URL url) {
 
             HttpConnection conn = Http.POST(sessionURL, "application/x-www-form-urlencoded");
             conn.setRequestBody(sessionRequestBody);
+
+            //when we request the session we need all interceptors except this one
+
+            conn.requestInterceptors.addAll(context.connection.requestInterceptors);
+            conn.requestInterceptors.remove(this);
+            conn.responseInterceptors.addAll(context.connection.responseInterceptors);
+            conn.responseInterceptors.remove(this);
```

Comment:
How would we know? We can introspect each interceptor, but they could be of any type. If they use the `CookieInterceptor` type then it will remove itself unless they override the behaviour. I don't think we can (currently) make any assumptions about the other interceptors and each has to take responsibility for how they execute any additional requests they do. I guess the alternative is generalizing this "remove self" behaviour in a utility (or making all interceptors extend an abstract type that provides it).

Did you have something in mind that we could do?